pkg/registries/registries.go

```diff
@@ -155,10 +187,32 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// If Prefix is a valid wildcard entry, we set scope to Prefix
+			if isValidWildcard(reg.Prefix) {
+				scope = reg.Prefix
+			}
+			if scopeMatchesRegistry(scope, blockedReg) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// isValidWildcard returns true if scope is a valid wildcard entry such as "*.registry.com"
+func isValidWildcard(scope string) bool {
```

Comment:
The way this function is used looks a bit problematic — it does a lot of input validation that should happen before we even get to `EditRegistriesConfig`. And at the point of `EditRegistriesConfig`, it returns `false` on various very invalid inputs, causing them to be treated as non-wildcards, which does not cause the outcome to be any more valid.

I think `EditRegistriesConfig` should either not be checking these details and assume valid wildcard format (and then just check for `scope[0] == '*' or something similarly simple; in that case something else should be validating before this function is called), or it should be failing.

It might well make sense for `runtime-utils` to publish a `ValidateRegistryWithPossibleWildcard` or something like that, to centralize the logic — that function should then return useful error messages instead of just `false`.

Reply:
Fixed.
Moved validation to the MCO code, so possible wildcard entries will be validated before EditRegistriesConfig will be called.